Source/Basic Shapes/SvgVisualElement.cs

```diff
@@ -132,50 +132,35 @@ protected override void Render(ISvgRenderer renderer)
         private void Render(ISvgRenderer renderer, bool renderFilter)
         {
             if (this.Visible && this.Displayable && this.PushTransforms(renderer) &&
-                (!Renderable || this.Path(renderer) != null))
+                (!this.Renderable || this.Path(renderer) != null))
             {
-                bool renderNormal = true;
-
-                if (renderFilter && this.Filter != null)
-                {
-                    var filterPath = this.Filter;
-                    if (filterPath.ToString().StartsWith("url("))
-                    {
-                        filterPath = new Uri(filterPath.ToString().Substring(4, filterPath.ToString().Length - 5), UriKind.RelativeOrAbsolute);
-                    }
-                    var filter = this.OwnerDocument.IdManager.GetElementById(filterPath) as FilterEffects.SvgFilter;
-                    if (filter != null)
-                    {
-                        this.PopTransforms(renderer);
-                        try
-                        {
-                            filter.ApplyFilter(this, renderer, (r) => this.Render(r, false));
-                        }
-                        catch (Exception ex) { Debug.Print(ex.ToString()); }
-                        renderNormal = false;
-                    }
-                }
-
-
-                if (renderNormal)
+                if (!(renderFilter && this.RenderFilter(renderer)))
                 {
                     this.SetClip(renderer);
 
-                    if (Renderable)
+                    if (this.Renderable)
                     {
-                        // If this element needs smoothing enabled turn anti-aliasing on
-                        if (this.RequiresSmoothRendering)
+                        var opacity = Math.Min(Math.Max(this.Opacity, 0), 1);
+                        if (opacity == 1f)
+                            this.RenderNormal(renderer);
+                        else
                         {
-                            renderer.SmoothingMode = SmoothingMode.AntiAlias;
-                        }
+                            IsPathDirty = true;

```

Comment:
Why this is needed for bounds calculation?

Reply:
`Bounds` property calls `Path(null)` method. (see `SvgPathBasedElement` class etc.)

Many `Path` methods keep path instance.
So call `IsPathDirty` property to recalculate including Stroke.